### DISCUSSION TITLE
A question about input and its shape

### DISCUSSION THREAD

**Discussion by Lastilla** [lululxvi/deepxde#1002] [Q&A] [answered]
@lululxvi，I'm so sorry to bother you. I use deepxde to solve the pde in this paper [s://asmedigitalcollection.asme.org/heattransfer/article/143/6/060801/1104439/Physics-Informed-Neural-Networks-for-Heat-Transfer](url) the pde as follows:
![image](https://user-images.githubusercontent.com/108338121/199921601-892f2b00-bee7-419e-8b12-fcf05e4a5ad6.png)

```
def pde(x, y):
    u = y[:, 0:1]
    v = y[:, 1:2]
    p = y[:, 2:3]
    theta = y[:, 3:4]
    du_x = dde.grad.jacobian(u, x, i=0, j=0)
    du_y = dde.grad.jacobian(u, x, i=0, j=1)
    dv_x = dde.grad.jacobian(v, x, i=0, j=0)
    dv_y = dde.grad.jacobian(v, x, i=0, j=1)
    dp_x = dde.grad.jacobian(p, x, i=0, j=0)
    dp_y = dde.grad.jacobian(p, x, i=0, j=1)
    dtheta_x = dde.grad.jacobian(theta, x, i=0, j=0)
    dtheta_y = dde.grad.jacobian(theta, x, i=0, j=1)
    du_xx = dde.grad.hessian(u, x, component=None, i=0, j=0)
    du_yy = dde.grad.hessian(u, x, component=None, i=0, j=1)
    dv_xx = dde.grad.hessian(v, x, component=None, i=0, j=0)
    dv_yy = dde.grad.hessian(v, x, component=None, i=0, j=1)
    dtheta_xx = dde.grad.hessian(theta, x, component=None, i=0, j=0)
    dtheta_yy = dde.grad.hessian(theta, x, component=None, i=0, j=1)
    continuity = du_x + dv_y
    x_momentum = C1 * (u * du_x + v * du_y) + dp_x - C2 * (du_xx + du_yy)
    y_momentum = C1 * (u * dv_x + v * dv_y) + dp_y - C2 * (dv_xx + dv_yy)
    x_theta = C3 * dtheta_xx - u * dtheta_x
    y_theta = C3 * dtheta_yy - v * dtheta_y
    return [continuity, x_momentum, y_momentum, x_theta, y_theta]
```
And the input layer can only feed a tensor of shape (?, 2), I can't figure out what did I input. My program is stuck now without any error.

**Comment by Lastilla** [comment, marked as answer]
My program can run but it is slow, that's my layer size`layer_size  = [2] + [60] * 12 + [5]`.

**Reply by haison19952013** [reply]
The layer size is quite small...If you have GPU, plz check if the driver is installed correctly..Or else, you can try Colab..

**Reply by Lastilla** [reply]
Thanks for replying. If I set the hidden layers shape as n*120, my GPU is incompetent. In this case, I just want to take a ez try. And the result about temperature is totally upside down.
![image](https://user-images.githubusercontent.com/108338121/199970454-d56d3d9d-4000-47a1-a6b1-95d56bcb3872.png)

**Reply by Lastilla** [reply]
![image](https://user-images.githubusercontent.com/108338121/199974927-6a51bf6b-0278-4014-a95f-f0c4004473d1.png)
Satisfactory results calculated on another computer
